0022-circuit-block.md

```diff
@@ -0,0 +1,359 @@
+As an annotation implies further processing by the quantum computer before execution by the QPU, the information on what annotations may be left in the circuit by the transpiler should be drawn from new information specified by the quantum computer, not the particular QPU.
+
+The transpiler should fail if a user or a pass in its pass manager inserts annotations on boxes that the backend does not support, and are not removed by the end of the transpilation.
+We need a way to communicate this information from backends to Qiskit SDK.
+
+For example, an advanced user might attempt to compile an `Estimator` pub with explicit IBM-specific markings of how the noise learning should be performed.
+The compilation of a pub represents partial compilation for a particular quantum computer, which will in turn perform further processing to turn this into the set of circuits that will actually be run on the QPU, and so Qiskit SDK's transpiler perhaps ought to have a similar specification of what constitutes "valid" output of such a compilation for quantum-computer processing, much like `Target` represents what constitues valid output for QPU processing.
+
+#### Very very rough sketch of how this might motivate a `BackendV3`
+
+> [!NOTE]
+> This is a _very vague_ sketch, I'm just trying to make sure we have a clear definition of what a `Target` is, and we don't start having it become so abstract that it is no longer a good model for a QPU.
+> This is absolutely not necessary to finalise for this RFC, but I feel like it's worth thinking about this as a joined-up story, since previously we have treated the output of `transpile` as "ready for QPU execution", but annotations stymie that.
+
+If we go this route, this might motivate a `BackendV3` definition which includes information on what annotations should be permitted to remain.
+This has some benefits:
+
+* the transpiler can reject the output from malformed pass-manager pipelines, or input circuits where the user put on a custom annotation that the backend does not know how to resolve.  This allows client-side validation of the circuits.
+* the QPY API could be extended to retrieve the serialisers/deserialisers from a `backend` object, rather than needing a user to construct it themselves.
+
+If we _truly_ associated the Qiskit object `Backend` with a "quantum computer" (as opposed to `Target`'s QPU), we might even consider an API where the primitives entry points are moved onto the `BackendV3` object, as in:
+
+```python
+class BackendV3:
+    @property
+    @abstractmethod
+    def target(self) -> Target: ...
+    @abstractmethod
+    def allowed_annotations(self) -> AnnotationChecker: ...
+    @abstractmethod
+    def sample(self, sampler_pub_likes) -> SamplerJob: ...
+    @abstractmethod
+    def estimate(self, estimator_pub_likes) -> EstimatorJob: ...
```

Comment:
My biggest concern here is perceived whiplash from users who were migrated away from `backend.run` to separate primitive classes, only to be ushered back to backend execution methods. 

I do agree, however, that having the correspondence QuantumComputer feature set <-> Backend, QPU feature set<-> Target is clean, is nice and clean.